src/psyclone/parse/module_info.py

```diff
@@ -92,6 +103,44 @@ def __init__(self, name, filename):
         # each module, indexed by the module names: Dict[str, Set(str)].
         self._used_symbols_from_module = None
 
+        # This variable will be a set that stores the name of all routines
+        # (based on fparser), # so we can test is a routine is defined
+        # without having to convert the AST to PSyIR. It is initialised with
+        # None so we avoid trying to parse a file more than once (parsing
+        # errors would cause routine_names to be empty, so we can test
+        # if routine_name is None vs if routine_names is empty)
+        self._routine_names = None
+
+        # This dictionary stores the mapping of routine name to the list
```

Comment:
Again, could we make this a property of `RoutineSymbol`, e.g. we could add a `get_schedule()` method to it?

Reply:
That also need support for interfaces to fully replace the current functionality, todo is there.

Reply:
TODO #2435 please.

Reply:
Done ... I think. The `_psyir_of_routines` has been removed, I don't know why it is still shown here. But I believe that todo also belongs to the `_generic_interfaces` below, so I have added it there.